torch_geometric/nn/aggr/basic.py

```diff
@@ -209,10 +224,20 @@ def forward(self, x: Tensor, index: Optional[Tensor] = None,
                 ptr: Optional[Tensor] = None, dim_size: Optional[int] = None,
                 dim: int = -2) -> Tensor:
 
+        if self.channels > 1:
+            self.assert_two_dimensional_input(x, dim)
+
+        p = self.p
+        if isinstance(p, Tensor):
+            shape = [1] * x.dim()
+            shape[dim + 1] = self.channels
+            p = p.view(*shape)
+
         out = self.reduce(x, index, ptr, dim_size, dim, reduce='mean')
-        if isinstance(self.p, (int, float)) and self.p == 1:
+
+        if isinstance(p, (int, float)) and p == 1:
```

Comment:
this logic is a bit odd to me... we skip the logic when `p = 1`? I  can see that its not very useful in that case, but I think it would be more consistent to still clamp personally

Reply:
I think the `clamp` is only needed to prevent overflows in `pow`. Not really sure what's the best approach here though so I left it as it is for now. Hope that works for you.

Reply:
Fair enough. Thanks